Audit/Persistance/Repositories/AuditLogRepository.cs

```diff
@@ -0,0 +1,120 @@
+        }
+
+        public Task<bool> Add(AuditLog entity)
+        {
+            _dbContext.AuditLogs.Add(entity);            
+            return Task.FromResult(_dbContext.SaveChanges() > 0);
+        }
+
+        public Task<AuditLog?> Get(AuditId id)
+        {
+            var log = _dbContext.AuditLogs.Find(id);
+            return Task.FromResult(log);
+        }
+
+        public Task<(IEnumerable<AuditLog>, PaginationMetadata)> GetByFacility(string facilityId, int pageSize, int pageNumber)
+        {
+            var logs = _dbContext.AuditLogs.Where(x => x.FacilityId == facilityId)
+                .Skip((pageNumber - 1) * pageSize)
+                .Take(pageSize)
+                .ToList();
+
+            var count = _dbContext.Set<AuditLog>().Count(x => x.FacilityId == facilityId);
+            PaginationMetadata metadata = new PaginationMetadata(pageSize, pageNumber, count);
+
+            return Task.FromResult<(IEnumerable<AuditLog>, PaginationMetadata)>((logs, metadata));
+        }
+
+        public Task<(IEnumerable<AuditLog>, PaginationMetadata)> Search(string? searchText, string? filterFacilityBy, string? filterCorrelationBy, string? filterServiceBy, string? filterActionBy, string? filterUserBy, string? sortBy, int pageSize, int pageNumber)
+        {
+            IEnumerable<AuditLog?> logs;
+            var query = _dbContext.AuditLogs.AsQueryable();
+
+            #region Build Query
+            if (searchText is not null && searchText.Length > 0)
+            {
+                query = query.Where(x =>
+                    (x.CorrelationId != null && x.CorrelationId.Contains(searchText)) ||
+                    (x.User != null && x.User.Contains(searchText)) ||
+                    (x.Notes != null && x.Notes.Contains(searchText)) ||
+                    (x.PropertyChanges != null && x.PropertyChanges.Any(y => y.NewPropertyValue.Contains(searchText)))
+                );
+            }
+
+            if (filterFacilityBy is not null && filterFacilityBy.Length > 0)
+            {
+                query = query.Where(x => x.FacilityId == filterFacilityBy);
+            }
+
+            if (filterCorrelationBy is not null && filterCorrelationBy.Length > 0)
+            {
+                query = query.Where(x => x.CorrelationId == filterCorrelationBy);
+            }
+
+            if (filterServiceBy is not null && filterServiceBy.Length > 0)
+            {
+                query = query.Where(x => x.ServiceName == filterServiceBy);
+            }
+
+            if (filterActionBy is not null && filterActionBy.Length > 0)
+            {
+                query = query.Where(x => x.Action == filterActionBy);
+            }
+
+            if (filterUserBy is not null && filterUserBy.Length > 0)
+            {
+                query = query.Where(x => x.User == filterUserBy);
+            }
+
+            if (sortBy is not null && sortBy.Length > 0)
+            {
+                query = sortBy switch
```

Comment:
Maybe apply a toLower() to the comparison if the search filters are publicly accessible?

Reply:
I think this is a good thing to look into. SQL Server is case insensitive by default, but if we switched providers to Postgres it would be case sensitive. Will create a ticket to look into the performance impact of this.